glib/src/variant.rs

```diff
@@ -178,7 +179,9 @@ impl Variant {
     ///
     /// Returns `Some` if self contains a `Variant`.
     #[inline]
-    pub fn variant(&self) -> Option<Variant> {
+    // FIXME reverted auto-rename, but the difference with `variant` is not obvious
+    // and this one looks more like a getter.
+    pub fn get_variant(&self) -> Option<Variant> {
```

Comment:
It's more a `try_get_as_variant()`, just like the `str()` one should be `try_get_as_str()` or similar (and same for array, etc below).

The `toml` crate calls these kind of functions `as_foo()` so let's maybe go with that?

Reply:
Also I don't know why these even exist. `variant.get::<glib::Variant>()` should do exactly the same if it was implemented? Let's keep them for now but rename into the style `toml` uses

Reply:
What do you think about the comment for the one just above:

```rust
    /// Boxes value.
    #[inline]
    // FIXME this one might be renamed as `new_variant`. See also below.
    pub fn variant(value: &Variant) -> Self {
        unsafe { from_glib_none(ffi::g_variant_new_variant(value.to_glib_none().0)) }
    }
```

Shall we move to `new_variant` which would follow Rust conventions better IMO? See also the changes we talked about before for the [`glib::ParamSpec::_type_`](https://github.com/fengalin/gtk-rs/blob/3f99b56b3941a5adbf67528d36d10b4b3d40fc18/glib/src/param_spec.rs#L168).

Reply:
I was talking about both. I think the correct thing to do here would be `fn from_variant(v: Variant) -> Variant` and `fn as_variant(&self) -> Option<Variant>` and the same for the other functions here.

Even more correct would be to just use the traits but they're incompletely implemented right now so that's for a later time.

Reply:
The paramspec case is a bit different because there it's actually about creating a new instance not *from* e.g. a string but *for* a string. `new_string()` or `for_string()` seems good there.

Reply:
Should be fixed now.

Reply:
Argh... Clippy doesn't like `as_*`: https://github.com/gtk-rs/gtk-rs/pull/211/checks?check_run_id=2314556400#step:21:99

Reply:
`as_*` for the getter, `from_*` for the constructor. array, tuple, maybe are constructors, that's why clippy is unhappy

Reply:
`form_*` doesn't seem right to me since it's the resulting type which is concerned by `*`. My suggestion would be these:

```rust
fn as_variant(&self) -> Option<Variant>;

fn new_as_variant(value: &Variant) -> Self;
fn new_as_array<T: StaticVariantType>(children: &[Variant]) -> Self;
fn new_as_tuple(children: &[Variant]) -> Self;
fn new_as_maybe<T: StaticVariantType>(child: Option<&Variant>) -> Self
```

Clippy is Ok with this :)

Reply:
Why `new_as`? It's creating a new `Variant` **from** an "array" (or rather slice in Rust terminogy), for example :)

Reply:
Let's take `new_as_tuple` as an example. It's also creating a variant from an `array` but I thought the point was that the resulting `Variant` was identified as being a `tuple`. Am I wrong?

Reply:
The tuple one only takes a slice currently because we don't have variadic tuples in Rust yet. Ideally it would take a `(Variant, Variant, Variant, ...)` or whatever the syntax for that would be.

Reply:
Oh, I see!